% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Assignment03},
  pdfauthor={Javad Meghrazi},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{Assignment03}
\author{Javad Meghrazi}
\date{4/3/2022}

\begin{document}
\maketitle

\hypertarget{introduction}{%
\subsection{Introduction}\label{introduction}}

In this article, we want to investigate the underlying genetics of
oviposition preference in two host races of planthopper
\emph{Nilaparvate nugens}. As a quantitative trait, the observed
variation in the trait could be attributed to 1) additive genetic
variance (\(V_A\)), which is related to the alleles who have an effect
on the trait regardless of the genetic background, 2) dominance genetic
variance (\(V_D\)), which is determined by the dominance coefficient of
the alleles contributing to this trait (if heterozygote's phenotype for
a locus is exactly equal to the mean of two homozygotes, then \(V_D\)
should be equal to 0), and 3) interaction genetic variance (\(V_I\))
which explains the effect of genetic background on the effect of each
allele on phenotype (taking into account the effect of all possible
interactions that different loci may have with each other). These 3
together explain the genetic variance (\(V_G\)) that is present in a
trait and also there is some variance that is due to environmental
factors (\(V_E\)) and is not inheritable. These variances have an
additive effect on the total variance that we observe in a trait in the
following way: \[ V_{total} = V_G + V_E\] \[ V_G = V_A + V_D + V_I\] The
goal of the following analysis is to understand the amount of variance
that could be attributed to each of the \(V_A\), \(V_D\), and \(V_E\).
In the assignment description, it implies that we want to know if the
genome proportion, dominance and interactions play a role in the trait,
but I think when it comes to quantitative traits, the underlying
genetics is usually complex and there is some \(V_D\) and \(V_I\) as
well as \(V_A\). For me, the question is how much are these variances
and I'll consider them as 0 if their contribution is so small that my
data is not enough to have a precise estimate of their importance. To
perform this task, first, we need to calculate the genomic proportion
that each line inherits from one of the pure lines as a measure of
additive genetic variance (\(V_A\)) and calculate the frequency of
heterozygotes in that line to see if dominance can explain any of the
observed patterns (because heterozygotes are the only place where the
effect of dominance can show itself).

\hypertarget{summary-of-the-data}{%
\subsection{Summary of the data}\label{summary-of-the-data}}

In the following graph and table, you can see the distribution, mean,
and sd of preference for different genotypes. Genotypes are arranged
according to the order of mean preference. As you can see the highest
preference is for rice and the lowest is for leer. F1 Hybrids of these
two lines have intermediate levels of preference which is what we
usually expect. Also, the hybrids of F1 and each of the lines have
intermediate preferences compared to the preference of F1 and that line.
What is interesting is that F2 hybrids have higher preferences than F1
which could tell us something about the underlying genetics (Probably
genetic interactions) and we should investigate it later. Also, F2
hybrids have a higher variance than F1 hybrids but that's something that
we expect based on mendelian genetics.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{hoppers }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{ (}\StringTok{"D:/courses\_2022\_sping/Quantitative methods\_ Dolph/BIOL\_501/Javad\_Assignment03/hopper.csv"}\NormalTok{) }\CommentTok{\# reading the data}
\NormalTok{hoppers}\SpecialCharTok{$}\NormalTok{genotype }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{ (hoppers}\SpecialCharTok{$}\NormalTok{genotype, }\AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"leer"}\NormalTok{, }\StringTok{\textquotesingle{}bl\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}f1\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}br\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}f2\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}rice\textquotesingle{}}\NormalTok{)) }\CommentTok{\# changing the order of genotypes in a way that}
\CommentTok{\#they are in the same order as their mean preference }
\FunctionTok{ggplot}\NormalTok{ (}\AttributeTok{data =}\NormalTok{ hoppers, }\FunctionTok{aes}\NormalTok{ (}\AttributeTok{x =}\NormalTok{ genotype, }\AttributeTok{y =}\NormalTok{ preference))}\SpecialCharTok{+} \CommentTok{\# ploting the data}
  \FunctionTok{geom\_boxplot}\NormalTok{()}\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{()}\SpecialCharTok{+} 
  \FunctionTok{labs}\NormalTok{ (}\AttributeTok{title =} \StringTok{"Oviposition preference for different genotypes"}\NormalTok{, }\AttributeTok{caption =} \StringTok{"bl is the hybrid of f1 and leer and br is the hybrid of f1 and rice"}\NormalTok{)}\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{ (}\AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{14}\NormalTok{),    }\CommentTok{\# Center title position and size}
    \AttributeTok{plot.subtitle =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{),            }\CommentTok{\# Center subtitle}
    \AttributeTok{plot.caption =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \DecValTok{0}\NormalTok{, }\AttributeTok{face =} \StringTok{"italic"}\NormalTok{))}\CommentTok{\# move caption to the left)}
\end{Highlighting}
\end{Shaded}

\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-2-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{hoppers }\OtherTok{\textless{}{-}}\NormalTok{ hoppers }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{mutate}\NormalTok{ (}\AttributeTok{genome\_proportion =} \FunctionTok{case\_when}\NormalTok{(genotype }\SpecialCharTok{==} \StringTok{"rice"} \SpecialCharTok{\textasciitilde{}} \DecValTok{1}\NormalTok{,}
\NormalTok{                                                             genotype }\SpecialCharTok{==} \StringTok{"leer"} \SpecialCharTok{\textasciitilde{}} \DecValTok{0}\NormalTok{, }
\NormalTok{                                                             genotype }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"f1"}\NormalTok{, }\StringTok{"f2"}\NormalTok{) }\SpecialCharTok{\textasciitilde{}} \FloatTok{0.5}\NormalTok{,}
\NormalTok{                                                             genotype }\SpecialCharTok{==} \StringTok{"br"} \SpecialCharTok{\textasciitilde{}} \FloatTok{0.75}\NormalTok{,}
\NormalTok{                                                             genotype }\SpecialCharTok{==} \StringTok{"bl"} \SpecialCharTok{\textasciitilde{}} \FloatTok{0.25}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%} \CommentTok{\# adding the genome proportion variable}
  \FunctionTok{mutate}\NormalTok{ (}\AttributeTok{dominance =} \FunctionTok{case\_when}\NormalTok{(genotype }\SpecialCharTok{==} \StringTok{"rice"} \SpecialCharTok{\textasciitilde{}} \DecValTok{0}\NormalTok{,}
\NormalTok{                                genotype }\SpecialCharTok{==} \StringTok{"leer"} \SpecialCharTok{\textasciitilde{}} \DecValTok{0}\NormalTok{, }
\NormalTok{                                genotype }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"f1"}\NormalTok{) }\SpecialCharTok{\textasciitilde{}} \DecValTok{1}\NormalTok{,}
\NormalTok{                                genotype }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"br"}\NormalTok{,}\StringTok{"bl"}\NormalTok{,}\StringTok{"f2"}\NormalTok{) }\SpecialCharTok{\textasciitilde{}} \FloatTok{0.5}\NormalTok{)) }\CommentTok{\# adding the dominance effect variable}

\NormalTok{hoppers }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{group\_by}\NormalTok{(genotype) }\SpecialCharTok{\%\textgreater{}\%} \CommentTok{\# creating the table}
  \FunctionTok{summarise}\NormalTok{ (}\AttributeTok{mean =} \FunctionTok{mean}\NormalTok{(preference) }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{round}\NormalTok{(}\AttributeTok{digits =} \DecValTok{2}\NormalTok{), }\AttributeTok{sd =} \FunctionTok{sd}\NormalTok{(preference) }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{round}\NormalTok{(}\AttributeTok{digits =} \DecValTok{2}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{kable}\NormalTok{(}\AttributeTok{caption =} \StringTok{"Oviposition preference of different genotypes"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{kable\_classic}\NormalTok{(}\AttributeTok{full\_width =}\NormalTok{ F, }\AttributeTok{html\_font =} \StringTok{"Cambria"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-3}Oviposition preference of different genotypes}
\centering
\begin{tabular}[t]{l|r|r}
\hline
genotype & mean & sd\\
\hline
leer & 0.09 & 0.82\\
\hline
bl & 0.46 & 1.07\\
\hline
f1 & 0.79 & 0.91\\
\hline
br & 1.20 & 1.08\\
\hline
f2 & 1.33 & 1.05\\
\hline
rice & 1.60 & 0.97\\
\hline
\end{tabular}
\end{table}

\hypertarget{models}{%
\subsection{Models:}\label{models}}

First I create a linear model to explain preference only based on genome
proportion

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gm\_genome }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{ (}\AttributeTok{data =}\NormalTok{ hoppers, preference }\SpecialCharTok{\textasciitilde{}}\NormalTok{ genome\_proportion) }\CommentTok{\# a inear model only based on genome prorportion}
\FunctionTok{summary}\NormalTok{ (gm\_genome) }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## lm(formula = preference ~ genome_proportion, data = hoppers)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -3.2230 -0.7199  0.0295  0.7195  3.0995 
## 
## Coefficients:
##                   Estimate Std. Error t value Pr(>|t|)    
## (Intercept)         0.2855     0.1239   2.304   0.0216 *  
## genome_proportion   1.3300     0.1927   6.904 1.47e-11 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.041 on 522 degrees of freedom
## Multiple R-squared:  0.08367,    Adjusted R-squared:  0.08191 
## F-statistic: 47.66 on 1 and 522 DF,  p-value: 1.475e-11
\end{verbatim}

Results of the summary show that genome proportion can explain 8.19\% of
the variation in the data (adjusted R-squared = 0.0819) and it's an
estimate of \(V_A\). Also, this model estimates the effect size of
genome proportion on preference to be 1.33. (Standard errors are in the
summary results but I don't include them in the text)

Next, I create a linear model to explain preference based on genome
proportion and dominance

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gm\_genome\_dom }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{(}\AttributeTok{data =}\NormalTok{ hoppers, preference }\SpecialCharTok{\textasciitilde{}}\NormalTok{ genome\_proportion }\SpecialCharTok{+}\NormalTok{ dominance) }\CommentTok{\# a linear model based on genome proportion and dominance effect}
\FunctionTok{summary}\NormalTok{ (gm\_genome\_dom)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## lm(formula = preference ~ genome_proportion + dominance, data = hoppers)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -3.2186 -0.7120  0.0335  0.7114  3.0935 
## 
## Coefficients:
##                   Estimate Std. Error t value Pr(>|t|)    
## (Intercept)         0.3780     0.1710   2.211   0.0275 *  
## genome_proportion   1.2884     0.1999   6.446 2.61e-10 ***
## dominance          -0.1315     0.1673  -0.786   0.4321    
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.041 on 521 degrees of freedom
## Multiple R-squared:  0.08475,    Adjusted R-squared:  0.08124 
## F-statistic: 24.12 on 2 and 521 DF,  p-value: 9.562e-11
\end{verbatim}

Results of the summary show that genome proportion and dominance effect
together can explain 8.12\% of the variation in the data (adjusted
R-squared = 0.0812) and it's an estimate of \(V_A + V_D\). Technically
it shouldn't be less than \(V_A\) but the fact that it's a bit smaller
than the previously calculated \(V_A\) shows that \(V_D\) is a small
number. The estimate of effect size of genome proportion is 1.29 and for
dominance it's -0.13. (Standard errors are in the summary results but I
don't include them in the text)

Next, I make a linear model to explain preference based on genotypes

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gm\_interaction }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{ (}\AttributeTok{data =}\NormalTok{ hoppers, preference }\SpecialCharTok{\textasciitilde{}}\NormalTok{ genotype) }\CommentTok{\# a linear model based on genotype}
\FunctionTok{summary}\NormalTok{ (gm\_interaction)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## lm(formula = preference ~ genotype, data = hoppers)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -3.14267 -0.70382  0.03474  0.68014  2.71733 
## 
## Coefficients:
##              Estimate Std. Error t value Pr(>|t|)    
## (Intercept)   0.09143    0.22346   0.409  0.68259    
## genotypebl    0.36838    0.26404   1.395  0.16356    
## genotypef1    0.69582    0.24790   2.807  0.00519 ** 
## genotypebr    1.11124    0.23542   4.720 3.04e-06 ***
## genotypef2    1.24185    0.24334   5.103 4.69e-07 ***
## genotyperice  1.51330    0.26267   5.761 1.43e-08 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.024 on 518 degrees of freedom
## Multiple R-squared:  0.1199, Adjusted R-squared:  0.1114 
## F-statistic: 14.11 on 5 and 518 DF,  p-value: 5.945e-13
\end{verbatim}

Results of the summary show that genotype can explain 11.14\% of the
variation in the data (adjusted R-squared = 0.114) and it's an estimate
of \(V_G\). Because the grouping variable is also genotype, this model
can explain all the variation that is attributable to genetics (\(V_G\))
(which is also all of the variation among different groups). The
unexplained variation is mainly due to environmental factors (\(V_E\)).
Even though this model explains a larger proportion of observed variance
than the other two models, we need to use a model selection algorithm to
compare the models. First I should decide whether I want to use BIC or
AIC.

\hypertarget{why-i-prefer-to-use-aic-over-bic}{%
\subsection{Why I prefer to use AIC over
BIC}\label{why-i-prefer-to-use-aic-over-bic}}

I know that AIC and BIC have different underlying assumptions, but I
think the most useful approach when we want to decide to use one is to
look at what they really do instead of looking at their assumptions.
Both AIC and BIC work by looking at the likelihood of the occurrence of
the results, but they penalize adding parameters in a different way. AIC
considers a constant penalty for adding each parameter while BIC
multiplies the number of parameters by the natural logarithm of sample
size. What BIC does is very effective when there are some explanatory
variables that might be totally irrelevant (specifically in cases when
we have many candidate explanatory variables). Under that scenario, if
we use AIC the number of present parameters in the best model increases
as we increase the sample size that we have. It happens because when we
have a larger sample size, we can have a better estimate of a larger
number of parameters and so the model chosen by AIC will have more
parameters, However, if there are parameters in the dataset that might
be irrelevant, if we have a very large sample size, there is a good
chance that AIC includes those parameters in the best model and we end
up over-fitting. On the other hand, BIC penalizes the addition of
parameters more heavily when there is a large sample size and there is a
lower chance of over-fitting (but a higher chance of under-fitting! it's
a trade-off).

In this case, the studied trait is a quantitative trait and our
explanatory variables are genotype, the proportion of genome received
from a line (an indicator of shared additive genetic variance), and
dominance. As I explained before, I believe when we are dealing with a
quantitative trait, the underlying genetics should be complex and there
is always some level of additive genetic elements, dominance and genetic
interaction present. The question is how important those components
(additive, dominance, and genetic interaction) are and whether our
sample size is large enough to give us a reliable measure of effect size
of each of the three components or not. Therefore, AIC should be used
because if BIC is used, it'll only look for strong effects and if there
is a small contribution by genetic interaction or dominance, it won't
capture it. On the other hand, there is no chance of over-fitting with
AIC as we know that all explanatory variables are important.~

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# calculating AIC for 3 models and comparing them }
\NormalTok{comparison }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}\AttributeTok{models  =} \FunctionTok{c}\NormalTok{ (}\StringTok{"gm\_genome"}\NormalTok{, }\StringTok{"gm\_genome\_dom"}\NormalTok{,}\StringTok{"gm\_interaction"}\NormalTok{),}
                         \AttributeTok{AIC =} \FunctionTok{c}\NormalTok{(}\FunctionTok{AIC}\NormalTok{(gm\_genome), }\FunctionTok{AIC}\NormalTok{ (gm\_genome\_dom), }\FunctionTok{AIC}\NormalTok{ (gm\_interaction))) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{mutate}\NormalTok{ (}\AttributeTok{deltaAIC =}\NormalTok{ AIC }\SpecialCharTok{{-}} \FunctionTok{min}\NormalTok{ (AIC))}
\NormalTok{comparison }\SpecialCharTok{\%\textgreater{}\%}   \FunctionTok{kable}\NormalTok{(}\AttributeTok{caption =} \StringTok{"Comparison of different models"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{kable\_classic}\NormalTok{(}\AttributeTok{full\_width =}\NormalTok{ F, }\AttributeTok{html\_font =} \StringTok{"Cambria"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-7}Comparison of different models}
\centering
\begin{tabular}[t]{l|r|r}
\hline
models & AIC & deltaAIC\\
\hline
gm\_genome & 1532.990 & 13.11797\\
\hline
gm\_genome\_dom & 1534.369 & 14.49678\\
\hline
gm\_interaction & 1519.872 & 0.00000\\
\hline
\end{tabular}
\end{table}

Based on the results the model with genotypes is the best model since it
has the lowest AIC score and as the difference in AIC of this model and
2 other models is more than 10, we can say that the data strongly
supports this model compared to the others. It's predictable that the
model based on genotypes would be better in AIC because genotype is the
grouping variable and it can explain all of the genetic variance. One
may think that results are suggesting that all of the genetic variance
could be attributed to interaction but that's not correct. I think in
this case we need a 4th model which explains the result based on
genome\_proportion and then tries to explain the remaining genetic
variation by adding genotypes. I'll explain it further.

\hypertarget{the-reason-that-i-included-the-4thh-model-genome_proportion-genotype}{%
\subsection{The reason that I included the 4thh model
(genome\_proportion +
genotype)}\label{the-reason-that-i-included-the-4thh-model-genome_proportion-genotype}}

In this data genotype is the grouping variable and therefore if we add
it as an explanatory variable, it can explain all of the variance that
exists among different groups. There is no doubt that the model
including the genotype will be the best model (though if there is no
significant interaction model based on genome proportion would be
closely good). However, by looking at the parameter estimates of the
model based on the genotype we cannot have an idea of the effect size of
genetic interactions. Therefore if we want to know the effect of genetic
interactions, we should have a model that tries to first explain the
variation using genome proportion as much as possible and then try to
explain the remaining variation using genotypes. Even though both of the
models (genotype and genome proportion + genotype) will have the same
AIC (because genotype is the grouping variable and there will not be any
unexplained between-group variance), the parameter estimates from the
genome proportion + genotype model give us a more accurate estimate of
the strength of genetic interactions.

\hypertarget{the-reason-that-i-included-a-null-model-intercept-only}{%
\subsection{The reason that I included a null model (intercept
only)}\label{the-reason-that-i-included-a-null-model-intercept-only}}

As the models based on genotypes and genome\_proportion + genotype are
going to give us the best fit, we should make a decision about including
genome\_proportion. To be able to know if genome proportion is
significantly improving the model, we should compare a model only based
on genome proportion to a model based on an intercept. If it has a much
better AIC score, then it's a good idea to choose the genome\_proportion
+ genotype model.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gm\_null }\OtherTok{\textless{}{-}}\NormalTok{ (}\FunctionTok{lm}\NormalTok{ (}\AttributeTok{data =}\NormalTok{ hoppers, preference }\SpecialCharTok{\textasciitilde{}} \DecValTok{1}\NormalTok{)) }\CommentTok{\# a linear model that fits preference based on an intercept}
\NormalTok{gm\_genome\_int }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{ (}\AttributeTok{data =}\NormalTok{ hoppers, preference }\SpecialCharTok{\textasciitilde{}}\NormalTok{ genome\_proportion }\SpecialCharTok{+}\NormalTok{ genotype) }\CommentTok{\# a linear model that first fits a line based on genome\_proportion and then add effect of interactions by adding genotypes}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# calculating AIC for 5 models and comparing them }
\NormalTok{comparison }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}\AttributeTok{models  =} \FunctionTok{c}\NormalTok{ (}\StringTok{"gm\_null"}\NormalTok{,}\StringTok{"gm\_genome"}\NormalTok{, }\StringTok{"gm\_genome\_dom"}\NormalTok{,}\StringTok{"gm\_interaction"}\NormalTok{, }\StringTok{"gm\_genome\_int"}\NormalTok{),}
                         \AttributeTok{AIC =} \FunctionTok{c}\NormalTok{(}\FunctionTok{AIC}\NormalTok{(gm\_null), }\FunctionTok{AIC}\NormalTok{(gm\_genome), }\FunctionTok{AIC}\NormalTok{ (gm\_genome\_dom), }\FunctionTok{AIC}\NormalTok{ (gm\_interaction), }\FunctionTok{AIC}\NormalTok{ (gm\_genome\_int))) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{mutate}\NormalTok{ (}\AttributeTok{deltaAIC =}\NormalTok{ AIC }\SpecialCharTok{{-}} \FunctionTok{min}\NormalTok{ (AIC))}
\NormalTok{comparison }\SpecialCharTok{\%\textgreater{}\%}   \FunctionTok{kable}\NormalTok{(}\AttributeTok{caption =} \StringTok{"Comparison of different models"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{kable\_classic}\NormalTok{(}\AttributeTok{full\_width =}\NormalTok{ F, }\AttributeTok{html\_font =} \StringTok{"Cambria"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-9}Comparison of different models}
\centering
\begin{tabular}[t]{l|r|r}
\hline
models & AIC & deltaAIC\\
\hline
gm\_null & 1576.776 & 56.90372\\
\hline
gm\_genome & 1532.990 & 13.11797\\
\hline
gm\_genome\_dom & 1534.369 & 14.49678\\
\hline
gm\_interaction & 1519.872 & 0.00000\\
\hline
gm\_genome\_int & 1519.872 & 0.00000\\
\hline
\end{tabular}
\end{table}

As we expect genome\_proportion + genotypes model similarly fits the
data as the model based on genotypes. Also, the model based on genome
proportion is much better than the null model and it shows that it's
important to have genome proportion in the final model. So I'm going to
choose genome\_proportion + genotype model as my final model.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{summary}\NormalTok{(gm\_genome\_int)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## lm(formula = preference ~ genome_proportion + genotype, data = hoppers)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -3.14267 -0.70382  0.03474  0.68014  2.71733 
## 
## Coefficients: (1 not defined because of singularities)
##                    Estimate Std. Error t value Pr(>|t|)    
## (Intercept)        0.091429   0.223456   0.409  0.68259    
## genome_proportion  1.513299   0.262674   5.761 1.43e-08 ***
## genotypebl        -0.009942   0.221502  -0.045  0.96422    
## genotypef1        -0.060825   0.169624  -0.359  0.72005    
## genotypebr        -0.023732   0.139050  -0.171  0.86455    
## genotypef2         0.485196   0.162877   2.979  0.00303 ** 
## genotyperice             NA         NA      NA       NA    
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.024 on 518 degrees of freedom
## Multiple R-squared:  0.1199, Adjusted R-squared:  0.1114 
## F-statistic: 14.11 on 5 and 518 DF,  p-value: 5.945e-13
\end{verbatim}

Results of the summary show that genotype can explain 11.14\% of the
variation in the data (adjusted R-squared = 0.114) and it's an estimate
of \(V_G\). We previously have an estimate of \(V_A\) (8.19\%) and we
know that the model with genome proportion is much better than the model
with only an intercept. So we can say that genome proportion is an
important parameter and we can find an estimate for its effect size from
the summary (the estimate is 1.51). The remaining variance (2.95\%)
could be attributed to genetic interactions (\(V_I\)) because we know
from AIC analysis that a model with interactions is better supported
than a model only based on genome proportion. Estimates of genotype
effect sizes are an indicator of the amount of interaction present for
each genotype. As you can see the estimate is close to 0 for all
genotypes except for F2. That makes sense because interactions arise
between different loci and F2 is the genotype where there is
heterogeneity in the origin of alleles in different loci. I mean in one
locus both alleles may have come from rice and in another locus, they
both may be from leer, and in another locus we may have one from each.
Therefore, it's where we expect to observe the effect of interactions
and we do (estimated effect size: 0.48).

Here I'm goiong to visualize the model fit

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{visreg}\NormalTok{ (gm\_genome\_int) }
\end{Highlighting}
\end{Shaded}

\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-11-1.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-11-2.pdf}

\hypertarget{comparison-with-conventional-null-hypothesis-testing}{%
\subsection{Comparison with conventional null hypothesis
testing}\label{comparison-with-conventional-null-hypothesis-testing}}

If I wanted to take a conventional null hypothesis approach, I should
have used a step-wise algorithm for the addition of different terms
(genome\_proportion, dominance, and genotype) to my model and that can
increase the type 1 error because of the multiple testing issue. Then,
if I wanted to adjust the alpha to avoid an increase in type 1 error,
the type 2 error would have increased. Overall, model selection is
superior to null hypothesis testing. provides higher power and also is
more precise.

\hypertarget{checking-the-assumptions-of-all-models}{%
\subsection{checking the assumptions of all
models}\label{checking-the-assumptions-of-all-models}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plot}\NormalTok{ (gm\_genome)}
\end{Highlighting}
\end{Shaded}

\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-1.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-2.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-3.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-4.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plot}\NormalTok{ (gm\_genome\_dom)}
\end{Highlighting}
\end{Shaded}

\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-5.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-6.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-7.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-8.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plot}\NormalTok{ (gm\_genome\_int)}
\end{Highlighting}
\end{Shaded}

\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-9.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-10.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-11.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-12.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plot}\NormalTok{ (gm\_interaction)}
\end{Highlighting}
\end{Shaded}

\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-13.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-14.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-15.pdf}
\includegraphics{MOHAMMADJAVAD.MEGHRAZI.ASSIGNMENT03_files/figure-latex/unnamed-chunk-12-16.pdf}
In all 4 models, the assumptions are met.

\end{document}
